examples/gmm/gmm.py

```diff
@@ -0,0 +1,168 @@
+    log_resp = weighted_log_prob - log_prob_norm[:, None]
+    return xp.mean(log_prob_norm), log_resp
+
+
+def m_step(X, xp, log_resp):
+    weights, means, covariances = \
+        estimate_gaussian_parameters(X, xp, xp.exp(log_resp))
+    inv_cov = 1 / xp.sqrt(covariances)
+    return weights, means, covariances, inv_cov
+
+
+def train_gmm(X, max_iter, tol):
+    xp = cupy.get_array_module(X)
+    lower_bound = -np.infty
+    converged = False
+    weights = xp.array([0.5, 0.5], dtype=np.float32)
+    mean1 = xp.random.normal(0, xp.array([1, 2]), size=2)
+    mean2 = xp.random.normal(-3, xp.array([2, 1]), size=2)
+    means = xp.stack((mean1, mean2))
+    covariances = xp.random.rand(2, 2)
+    inv_cov = 1 / xp.sqrt(covariances)
+
+    for n_iter in six.moves.range(max_iter):
+        prev_lower_bound = lower_bound
+        log_prob_norm, log_resp = e_step(X, xp, inv_cov, means, weights)
+        weights, means, covariances, inv_cov = m_step(X, xp, log_resp)
+        lower_bound = log_prob_norm
+        change = lower_bound - prev_lower_bound
+        if abs(change) < tol:
+            converged = True
+            break
+
+    if not converged:
+        msg = 'Failed to converge. Increase max-iter or tol.'
+        print(msg)
+
+    return inv_cov, means, weights, covariances
+
+
+def predict(X, inv_cov, means, weights):
+    xp = cupy.get_array_module(X)
+    log_prob = estimate_log_prob(X, xp, inv_cov, means)
+    return (log_prob + xp.log(weights)).argmax(axis=1)
+
+
+def calc_acc(X_train, y_train, X_test, y_test, max_iter, tol):
+    xp = cupy.get_array_module(X_train)
+    inv_cov, means, weights, cov = train_gmm(X_train, max_iter, tol)
+    y_train_pred = predict(X_train, inv_cov, means, weights)
+    train_accuracy = xp.mean(y_train_pred == y_train) * 100
+    y_test_pred = predict(X_test, inv_cov, means, weights)
+    test_accuracy = xp.mean(y_test_pred == y_test) * 100
+    print('train_accuracy : %f' % train_accuracy)
+    print('test_accuracy : %f' % test_accuracy)
+    return y_test_pred, means, cov
+
+
+def draw(X, pred, means, covariances, output):
+    xp = cupy.get_array_module(X)
+    for i in six.moves.range(2):
+        labels = X[pred == i]
+        if xp == cupy:
+            labels = labels.get()
+        plt.scatter(labels[:, 0], labels[:, 1], color=np.random.rand(3, 1))
+    if xp == cupy:
+        means = means.get()
+        covariances = covariances.get()
+    plt.scatter(means[:, 0], means[:, 1], s=120, marker='s', facecolors='y',
+                edgecolors='k')
+    x = np.linspace(-10, 10, 1000)
+    y = np.linspace(-10, 10, 1000)
+    X, Y = np.meshgrid(x, y)
+    for i in six.moves.range(2):
+        Z = mlab.bivariate_normal(X, Y, np.sqrt(covariances[i][0]),
+                                  np.sqrt(covariances[i][1]),
+                                  means[i][0], means[i][1])
+        plt.contour(X, Y, Z)
+    plt.savefig(output + '.png')
+
+
+def run(gpuid, max_iter, tol, output):
```

Comment:
How about printing some header line before each of trials (CPU/GPU)? Such as Running CPU... or something like that. It's helpful when an error occurred or not converged etc.